src/lib/routes.js

```diff
@@ -1,50 +1,17 @@
 'use strict';
 
-const dateFormat = 'YYYY-MM-DD HH:mm:ss';
-const moment = require('moment');
-const model = require('./model');
-const dashboardModel = require('./dashboard-model');
+const { promiseResponder, dashboard, home } = require('./route-helpers');
 const { LevDashboard, LevReport } = require('lev-react-components');
 
-const promiseResponder = (promise, req, res, next, component) => promise
-  .then(data => component ? res.render(component, data) : res.send(data))
-  .catch(err => {
-    req.log.error(err);
-    return next(err);
-  });
-
-const dateChecker = d => !!d && moment(d, dateFormat).isValid() && d;
-
 module.exports = server => {
   server.get('/readiness', (req, res) => res.send('OK'));
 
-  server.get('/data', (req, res, next) => { // eslint-disable-line consistent-return
-    const fromDate = dateChecker(req.query.from);
-    const toDate = dateChecker(req.query && req.query.to);
-
-    if (dateChecker(fromDate)) {
-      if (dateChecker(toDate)) {
-        return promiseResponder(model(fromDate, toDate), req, res, next);
-      }
-      return next(new server.errors.BadRequestError(`Make sure the date format is "${dateFormat}" (time is optional)`));
-    }
-    return next(new server.errors.BadRequestError('Must provide "from" date parameter, and optionally a "to" date'));
-  });
-
-  server.get('/dashboard', (req, res, next) => promiseResponder(dashboardModel(), req, res, next, LevDashboard));
+  server.get('/data', (req, res, next) =>
+    promiseResponder(home(req.query, server.errors.BadRequestError), req, res, next));
 
-  server.get('/dashboard/data', (req, res, next) => promiseResponder(dashboardModel(), req, res, next));
+  server.get('/dashboard', (req, res, next) => promiseResponder(dashboard(), req, res, next, LevDashboard));
 
-  server.get('/*', (req, res, next) => {
-    const fromDate = dateChecker(req.query && req.query.from);
-    const toDate = dateChecker(req.query && req.query.to);
-    const searchGroup = req.query && req.query.currentGroup;
+  server.get('/dashboard/data', (req, res, next) => promiseResponder(dashboard(), req, res, next));
```

Comment:
FWIW, `promiseResponder` could be rewritten to have the following interface:
```js
const promiseResponder = (promise, Component) => (req, res, next) => {
  [...]
```
That way when called it could simply be:
```js
server.get('/dashboard/data', promiseResponder(dashboard());
```

Reply:
It's not a big deal though and doesn't need to be done now, unless you wanted to.

Reply:
Updated the code to reflect this